tests/common/src/pkcs11/pkcs11.cmake

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
+# Copyright (C) 2024 EPAM Systems, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# ######################################################################################################################
+# Setup SoftHSM2
+# ######################################################################################################################
+
+find_library(
+    SOFTHSM2_LIB
+    NAMES softhsm2
+    PATH_SUFFIXES softhsm
+)
+
+if(NOT SOFTHSM2_LIB)
+    message(FATAL_ERROR "softhsm2 library not found")
+endif()
+
+set(SOFTHSM_BASE_DIR ${CMAKE_CURRENT_BINARY_DIR}/softhsm)
+
+file(MAKE_DIRECTORY ${SOFTHSM_BASE_DIR}/tokens/)
+
+configure_file(${CMAKE_CURRENT_LIST_DIR}/softhsm2.conf ${SOFTHSM_BASE_DIR}/softhsm2.conf COPYONLY)
+file(APPEND ${SOFTHSM_BASE_DIR}/softhsm2.conf "directories.tokendir = ${SOFTHSM_BASE_DIR}/tokens/\n")
+
+# ######################################################################################################################
+# Create PKCS11 test certificates
+# ######################################################################################################################
+
+set(CERTIFICATES_DIR ${CMAKE_CURRENT_BINARY_DIR}/certificates)
+
+file(MAKE_DIRECTORY ${CERTIFICATES_DIR})
+
+message("\nGenerating PKCS11 test certificates...")
+
+message("  Create a Certificate Authority private key...")
+execute_process(
+    COMMAND
+        openssl req -new -newkey rsa:2048 -nodes -out ${CERTIFICATES_DIR}/ca.csr -keyout ${CERTIFICATES_DIR}/ca.key
+        -nodes -subj "/C=XX/ST=StateName/L=CityName/O=Epam/OU=CompanySectionName/CN=Aos Core" COMMAND_ERROR_IS_FATAL
+        ANY
+)
+
+message("  Create a CA self-signed certificate:")
+execute_process(
+    COMMAND openssl x509 -trustout -signkey ${CERTIFICATES_DIR}/ca.key -days 365 -req -in ${CERTIFICATES_DIR}/ca.csr
+            -out ${CERTIFICATES_DIR}/ca.pem COMMAND_ERROR_IS_FATAL ANY
+)
+
+message(
+    "  Issue a client certificate by first generating the key, then request, then sign the certificate using private key of the CA..."
```

Comment:
Ditto and `Issue a client certificate` message would be enough

Reply:
fixed